frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-popover.tsx

```diff
@@ -0,0 +1,128 @@
+    switch (status) {
+      case UPLOAD_STATUS.ERROR:
+        return {
+          title: `Upload Error`,
+          body: error?.message,
+          icon: <ErrorCircleOIcon className="co-icon-and-text__icon" color="#cf1010" />,
+        };
+      case UPLOAD_STATUS.CANCELED:
+        return {
+          title: error ? 'Cancel Error' : 'Upload Cancelling',
+          body: error ? error?.message : 'Removing Resources',
+          icon: (
+            <ErrorCircleOIcon className="co-icon-and-text__icon" color={error ? '#cf1010' : ''} />
+          ),
+        };
+      case UPLOAD_STATUS.UPLOADING:
+        return {
+          title: `Uploading`,
+          body: 'Please do not close this window, you can keep navigating the app freely.',
+          icon: <InProgressIcon className="co-icon-and-text__icon" />,
+        };
+      case UPLOAD_STATUS.SUCCESS:
+        return {
+          title: `Upload Finished`,
+          icon: <InProgressIcon className="co-icon-and-text__icon" />,
+        };
+      default:
+        return null;
+    }
+  };
+
+  if (upload) {
+    const uploadPopoverBody = getPopoverBody(upload?.uploadStatus);
+    return (
+      <div key={pvcName} className="kv__upload-status">
+        {uploadPopoverBody?.icon}
+        <Popover
+          headerContent={<div>{uploadPopoverBody?.title}</div>}
+          position={PopoverPosition.bottom}
+          className="kv-pvc-upload__popover"
+          bodyContent={
+            <Stack hasGutter>
+              <StackItem>
+                {upload?.uploadStatus === UPLOAD_STATUS.CANCELED && (
+                  <Spinner size="md" className="co-icon-and-text__icon" />
+                )}
+                {uploadPopoverBody?.body}
+              </StackItem>
+              <StackItem>
+                <Progress
+                  value={upload?.progress}
+                  title={upload?.fileName}
+                  variant={getProgressVariant(upload?.uploadStatus)}
+                />
+              </StackItem>
+              {upload?.uploadStatus === UPLOAD_STATUS.UPLOADING && (
+                <StackItem>
+                  <Button
+                    id="cdi-upload-cancel-btn"
+                    className="pf-m-link--align-left"
+                    variant="link"
+                    onClick={() => {
+                      upload.cancelUpload();
+                      killUploadPVC(pvcName, namespace).catch((err) => setError(err));
+                    }}
+                  >
+                    Cancel upload
+                  </Button>
+                </StackItem>
+              )}
+            </Stack>
+          }
+        >
+          <Button id="cdi-upload-popover-btn" className="pf-m-link--align-left" variant="link">
+            {uploadPopoverBody?.title}
+          </Button>
+        </Popover>
+      </div>
+    );
+  }
+  // no context data, upload is from another session
+  return <StatusIconAndText icon={<InProgressIcon />} title="Uploading" />;
```

Comment:
you can use `ProgressStatus`

```suggestion
  return <ProgressStatus title="Uploading" />;
```